Income form

AKidd95 opened the pull request:
#### PT Story: https://www.pivotaltracker.com/story/show/155546556
#### Description

* Adds income form
* Pass all acceptance tests  
* Adds flash messages for successful entry

#### What I have learned working on this feature: 
* How to generate controller and models
* Proper routing techniques
* How to create forms using form_with

AKidd95 committed:
WIP user enters monthly income
create routes for home controller
set submit form on index page

AKidd95 committed:
create income model
create show page
create income db
add resources to routes

isabellelii committed:
Merge branch 'income_form' of https://github.com/AKidd95/broke_no_more into income_form

isabellelii committed:
WIP, try to fix error

AKidd95 committed:
add income controller
add income_new page
add income_new route

AKidd95 committed:
Add flash keys to controller and view.
fix form format
all tests pass

AKidd95 committed:
clean up rails_helper

AKidd95 committed:
change spec to rspec

AKidd95 committed:
sanitize income params
create rspec tests for income model

magnus-thor commented on app/controllers/home_controller.rb:
```diff
@@ -1,4 +1,17 @@
+
+  def update
+    binding.pry
+  end
```
Remember to remove Binding.pry's
remove methods that you are not using

magnus-thor commented on app/controllers/income_controller.rb:
```diff
@@ -0,0 +1,26 @@
+  end
+
+  def update
+  end
```
Is this being used?

magnus-thor commented on features/income_form.feature:
```diff
@@ -0,0 +1,12 @@
+  Scenario: User enters their monthly income
+  Given he is on home page
+  Then he clicks "New Income"
+  And he sees "Enter Monthly Income" form
```
Use the `he should see` step to keep your code dry

isabellelii committed:
Deleted income controller
created Incomes controller
Deleted unnessecery binding prys
Deleted Url from form

isabellelii committed:
Deleted unnessecary methods in controller
Changed feature file

magnus-thor commented on app/views/incomes/new.html.erb:
```diff
@@ -0,0 +1,5 @@
+<%= form_with model: Income.new do |form| %>
+  <%= "Enter Monthly Income" %>
+  <%= form.number_field :amount, :placeholder => "Fill here" %>
```
Old syntax, do `placeholder: 'Fill here'`, and use single qoutes

magnus-thor commented on features/step_definitions/income_form_steps.rb:
```diff
@@ -0,0 +1,15 @@
+end
+
+Given("he fills in {string} with {string}") do |input, value|
+  fill_in(input, :with => value)
```
Old syntax, use `with: value`

AKidd95 committed:
clean up old syntax
fix merge conflicts

magnus-thor reviewed (approved):
Looks good
